File: iosApp/iosApp/models/Burger.swift
import Foundation

/**
 * 2.1. Create the Burger model natively
 */
struct Burger: Identifiable {
    let id: Int
    let name: String
    let description: String
    let price: Float
    let thumbnailName: String
    
    init(id: Int, name: String, description: String, price: Float, thumbnailName: String) {
        self.id = id
        self.name = name
        self.description = description
        self.price = price
        self.thumbnailName = thumbnailName
    }
}

/**
 * 3.1.1. Play with the Burgers data using a static list (Native)
 */
extension Burger {
    static var burgers: [Burger] {
        [
            Burger(
                id: 0,
                name: "Beef Burger",
                description: "Onion with cheese",
                price: 18,
                thumbnailName: "BeefBurgerThumbnail"
            ),
            Burger(
                id: 1,
                name: "Chicken Burger",
                description: "Cheese with chicken",
                price: 12,
                thumbnailName: "ChickenBurgerThumbnail"
            ),
            Burger(
                id: 2,
                name: "Classic Burger",
                description: "Beef with lettuce",
                price: 24,
                thumbnailName: "ClassicBurgerThumbnail"
            ),
            Burger(
                id: 3,
                name: "Grilled Burger",
                description: "Grilled chicken",
                price: 14,
                thumbnailName: "GrilledBurgerThumbnail"
            )
        ]
    }
}

File: androidApp/src/main/java/io/github/tiagonuneslx/burgerapp/android/db/entity/Burger.kt
package io.github.tiagonuneslx.burgerapp.android.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
@Entity
data class Burger(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "rowid") val id: Int = 0,
    val name: String,
    val description: String,
    val price: Float,
    val thumbnailUrl: String,
) {
    companion object {
        val samples = listOf(
            Burger(
                1,
                "Beef Burger",
                "Onion with cheese",
                18f,
                "/static/beef_burger_thumbnail.png"
            ),
            Burger(
                2,
                "Chicken Burger",
                "Cheese with chicken",
                12f,
                "/static/chicken_burger_thumbnail.png"
            ),
            Burger(
                3,
                "Classic Burger",
                "Beef with lettuce",
                24f,
                "/static/classic_burger_thumbnail.png"
            ),
            Burger(
                4,
                "Grilled Burger",
                "Grilled chicken",
                14f,
                "/static/grilled_burger_thumbnail.png"
            ),
        )
    }
}